store/sqlstore/compliance_store.go

```diff
@@ -36,226 +36,210 @@ func NewSqlComplianceStore(sqlStore SqlStore) store.ComplianceStore {
-			result.Err = model.NewAppError("SqlComplianceStore.Get", "store.sql_compliance.get.finding.app_error", nil, err.Error(), http.StatusInternalServerError)
-		} else {
-			result.Data = compliances
-		}
-	})
+	var compliances model.Compliances
+	if _, err := s.GetReplica().Select(&compliances, query, map[string]interface{}{"Offset": offset, "Limit": limit}); err != nil {
+		return nil, model.NewAppError("SqlComplianceStore.Get", "store.sql_compliance.get.finding.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+	return compliances, nil
 }
 
-func (us SqlComplianceStore) Get(id string) store.StoreChannel {
-	return store.Do(func(result *store.StoreResult) {
-		if obj, err := us.GetReplica().Get(model.Compliance{}, id); err != nil {
-			result.Err = model.NewAppError("SqlComplianceStore.Get", "store.sql_compliance.get.finding.app_error", nil, err.Error(), http.StatusInternalServerError)
-		} else if obj == nil {
-			result.Err = model.NewAppError("SqlComplianceStore.Get", "store.sql_compliance.get.finding.app_error", nil, err.Error(), http.StatusNotFound)
-		} else {
-			result.Data = obj.(*model.Compliance)
-		}
-	})
+func (us SqlComplianceStore) Get(id string) (*model.Compliance, *model.AppError) {
+	obj, err := us.GetReplica().Get(model.Compliance{}, id)
+	if err != nil {
+		return nil, model.NewAppError("SqlComplianceStore.Get", "store.sql_compliance.get.finding.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+	if obj == nil {
+		return nil, model.NewAppError("SqlComplianceStore.Get", "store.sql_compliance.get.finding.app_error", nil, err.Error(), http.StatusNotFound)
+	}
+	return obj.(*model.Compliance), nil
 }
 
-func (s SqlComplianceStore) ComplianceExport(job *model.Compliance) store.StoreChannel {
-	return store.Do(func(result *store.StoreResult) {
-		props := map[string]interface{}{"StartTime": job.StartAt, "EndTime": job.EndAt}
-
-		keywordQuery := ""
-		keywords := strings.Fields(strings.TrimSpace(strings.ToLower(strings.Replace(job.Keywords, ",", " ", -1))))
-		if len(keywords) > 0 {
+func (s SqlComplianceStore) ComplianceExport(job *model.Compliance) ([]*model.CompliancePost, *model.AppError) {
+	props := map[string]interface{}{"StartTime": job.StartAt, "EndTime": job.EndAt}
 
-			keywordQuery = "AND ("
+	keywordQuery := ""
+	keywords := strings.Fields(strings.TrimSpace(strings.ToLower(strings.Replace(job.Keywords, ",", " ", -1))))
+	if len(keywords) > 0 {
 
-			for index, keyword := range keywords {
-				if index >= 1 {
-					keywordQuery += " OR LOWER(Posts.Message) LIKE :Keyword" + strconv.Itoa(index)
-				} else {
-					keywordQuery += "LOWER(Posts.Message) LIKE :Keyword" + strconv.Itoa(index)
-				}
+		keywordQuery = "AND ("
 
-				props["Keyword"+strconv.Itoa(index)] = "%" + keyword + "%"
+		for index, keyword := range keywords {
+			if index >= 1 {
+				keywordQuery += " OR LOWER(Posts.Message) LIKE :Keyword" + strconv.Itoa(index)
```

Comment:
Unrelated to this PR, but I think the Compliance Store would benefit greatly from the SQL Builder (Squirrel) as well.

Reply:
Yes, but that would be a mayor change, I really want to start using squirrel every where, but I going to defer it to other PR in the future.